engines/wintermute/detection.cpp

```diff
@@ -210,6 +214,1421 @@ class WintermuteMetaEngine : public AdvancedMetaEngine {
+			act->addDefaultInputMapping("JOY_DOWN"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SHIFT", _("Shift"));
+			act->setKeyEvent(KEYCODE_RSHIFT);
+			act->addDefaultInputMapping("LSHIFT"); // original keyboard control
+			act->addDefaultInputMapping("RSHIFT"); // original keyboard control
+			act->addDefaultInputMapping("MOUSE_MIDDLE"); // extra mouse
+			act->addDefaultInputMapping("JOY_Y"); // extra joy
+			gameKeyMap->addAction(act);
+		} else if (gameId == "palladion") {
+			act = new Action("HINT", _("Show hints"));
+			act->setKeyEvent(KeyState(KEYCODE_SPACE, ASCII_SPACE));
+			act->addDefaultInputMapping("SPACE"); // original keyboard
+			act->addDefaultInputMapping("MOUSE_MIDDLE"); // extra mouse
+			act->addDefaultInputMapping("JOY_Y"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRS", _("Save screenshot"));
+			act->setKeyEvent(KEYCODE_F12);
+			act->addDefaultInputMapping("F12"); // original keyboard
+			gameKeyMap->addAction(act);
+		} else if (gameId == "papasdaughters1") {
+			act = new Action("SCRLUP", _("Scroll up"));
+			act->setMouseWheelUpEvent();
+			act->addDefaultInputMapping("MOUSE_WHEEL_UP"); // original mouse
+			act->addDefaultInputMapping("UP"); // extra keyboard
+			act->addDefaultInputMapping("JOY_UP"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRLDN", _("Scroll down"));
+			act->setMouseWheelDownEvent();
+			act->addDefaultInputMapping("MOUSE_WHEEL_DOWN"); // original mouse
+			act->addDefaultInputMapping("DOWN"); // extra keyboard
+			act->addDefaultInputMapping("JOY_DOWN"); // extra joy
+			gameKeyMap->addAction(act);
+		} else if (gameId == "papasdaughters2") {
+			act = new Action("HINT", _("Show hints"));
+			act->setKeyEvent(KEYCODE_F1);
+			act->addDefaultInputMapping("F1"); // original keyboard
+			act->addDefaultInputMapping("MOUSE_MIDDLE"); // extra mouse
+			act->addDefaultInputMapping("JOY_Y"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRLUP", _("Scroll up"));
+			act->setMouseWheelUpEvent();
+			act->addDefaultInputMapping("MOUSE_WHEEL_UP"); // original mouse
+			act->addDefaultInputMapping("UP"); // extra keyboard
+			act->addDefaultInputMapping("JOY_UP"); // extra joy
+			gameKeyMap->addAction(act);
+
+			act = new Action("SCRLDN", _("Scroll down"));
+			act->setMouseWheelDownEvent();
+			act->addDefaultInputMapping("MOUSE_WHEEL_DOWN"); // original mouse
+			act->addDefaultInputMapping("DOWN"); // extra keyboard
+			act->addDefaultInputMapping("JOY_DOWN"); // extra joy
+			gameKeyMap->addAction(act);
+		} else if (gameId == "pizzamorgana") {
+			act = new Action("ACTNXT", _("Next action"));
+			act->setRightClickEvent();
+			act->addDefaultInputMapping("MOUSE_RIGHT"); // original mouse (globally)
+			act->addDefaultInputMapping("LEFTBRACKET"); // original keyboard (ingame)
+			act->addDefaultInputMapping("MOUSE_WHEEL_UP"); // original mouse
```

Comment:
This doesn't seem to match what is described on the [Wiki article](https://wiki.scummvm.org/index.php?title=Wintermute/Controls#Pizza_Morgana:_Episode_1_-_Monsters_and_Manipulations_in_the_Magical_Forest), which lists "[" for selecting the previous action and "]" for selecting the next action.

Reply:
Thanks for pointing this. Took a deep look at Pizza Morgana and ZBang code.
Wiki version was correct, both those games share code that handles MouseWheelUp, RightClick, RightDoubleClick and "]" as "Select Next Action", and MouseWheelDown & "[" as "Select Previous Action". Fixed.